gammapy/irf/psf_king.py

```diff
@@ -84,44 +96,91 @@ def from_table(cls, table):
 
         energy_lo = table['ENERG_LO'].squeeze()
         energy_hi = table['ENERG_HI'].squeeze()
-        energy = np.sqrt(energy_lo * energy_hi)
-        energy = Energy(energy, unit=table['ENERG_LO'].unit)
+        energy_lo = Energy(energy_lo, unit=table['ENERG_LO'].unit)
+        energy_hi = Energy(energy_hi, unit=table['ENERG_HI'].unit)
 
         gamma = Quantity(table['GAMMA'].squeeze(), table['GAMMA'].unit)
         sigma = Quantity(table['SIGMA'].squeeze(), table['SIGMA'].unit)
 
-        return cls(offset, energy, gamma, sigma)
+        try:
+            energy_thresh_lo = Quantity(table.meta['LO_THRES'], 'TeV')
+            energy_thresh_hi = Quantity(table.meta['HI_THRES'], 'TeV')
+            return cls(energy_lo, energy_hi, offset, gamma, sigma, energy_thresh_lo, energy_thresh_hi)
+        except KeyError:
+            log.warning('No safe energy thresholds found. Setting to default')
+            return cls(energy_lo, energy_hi, offset, gamma, sigma)
+
+    def to_fits(self):
+        """
+        Convert psf table data to FITS hdu list.
+
+        Returns
+        -------
+        hdu_list : `~astropy.io.fits.HDUList`
+            PSF in HDU list format.
+        """
+        # Set up data
+        names = ['ENERG_LO', 'ENERG_HI', 'THETA_LO', 'THETA_HI',
+                 'SIGMA', 'GAMMA']
+        units = ['TeV', 'TeV', 'deg', 'deg',
+                 'deg', '']
+        data = [self.energy_lo, self.energy_hi, self.offset, self.offset,
+                self.sigma, self.gamma]
+
+        table = Table()
```

Comment:
if you want to you can split this out into a ``to_table`` method, but I don't think it's necessary at the moment